### PR TITLE
PP-2446 product client

### PR DESCRIPTION
**Pull request by kakumara**
## CONTEXT
Product is a selfservice representation of the `Product` concept from products microservice. Its identical except for not including a `pay_api_key` (purposely).

## WHATS INCLUDED
- Similar to other external integrations introduced `products_client` which is a wrapper for `products` microservice interactions.
- `products_service.createProduct` orchestrates between `publicAuth` and `products` in order to create a product successfully
- The client of this method must make sure to perform necessary validations in `productOptions` to include all mandatory fields required

Other than that
  - added Pact based integration tests
  - added product_fixtures

## NOT DONE
- the controller and UI's for creating prototype payments in not included in this PR. (next to come)

### CODE REVIEW

File: app/models/Product.class.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+'use strict'
+
+/**
+ @class Product
+ */
+class Product {
```
Comment (maxcbc) [inline app/models/Product.class.js]: Noice

File: app/services/clients/products_client.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
+'use strict'
+
+const q = require('q')
```
Comment (maxcbc) [inline app/services/clients/products_client.js]: We're switching to using native Promises on frontend for performance. 

We ought to do the same here. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise

File: app/models/Product.class.js (hunk 1 above)
Comment (kakumara) [reply app/models/Product.class.js]: 👍

File: app/services/clients/products_client.js (hunk 2 above)
Comment (kakumara) [reply app/services/clients/products_client.js]: Agree. But we'll have to look at this later as changing this is a bit `convoluted` and causes all other client changes causing bit heavy recfactoring. Prehaps should do this as a separate effort. I'll adopt it in the new node microservice

Comment (maxcbc) [reply app/services/clients/products_client.js]: OK

**Pull request merged**